ibm_cloud_sdk_core/base_service.py

```diff
@@ -22,284 +22,100 @@
-                if self.ICP4D_ACCESS_TOKEN in self.vcap_service_credentials:
-                    self.set_icp4d_access_token(self.vcap_service_credentials.get(self.ICP4D_ACCESS_TOKEN))
-
-        if (self.username is None or self.password is None) and self.token_manager is None:
-            raise ValueError(
-                'You must specify your IAM api key or username and password service '
-                'credentials (Note: these are different from your IBM Cloud id)')
 
-    def _load_from_credential_file(self, service_name, separator='='):
-        """
-        Initiates the credentials based on the credential file
+        self._set_user_agent_header(self._build_user_agent())
 
-        :param str service_name: The service name
-        :param str separator: the separator for key value pair
-        """
-        # File path specified by an env variable
-        credential_file_path = os.getenv('IBM_CREDENTIALS_FILE')
-
-        # Home directory
-        if credential_file_path is None:
-            file_path = join(expanduser('~'), self.DEFAULT_CREDENTIALS_FILE_NAME)
-            if isfile(file_path):
-                credential_file_path = file_path
-
-        # Top-level of the project directory
-        if credential_file_path is None:
-            file_path = join(dirname(dirname(abspath(__file__))), self.DEFAULT_CREDENTIALS_FILE_NAME)
-            if isfile(file_path):
-                credential_file_path = file_path
-
-        if credential_file_path is not None:
-            with open(credential_file_path, 'r') as fp:
-                for line in fp:
-                    key_val = line.strip().split(separator)
-                    if len(key_val) == 2:
-                        self._set_credential_based_on_type(service_name, key_val[0].lower(), key_val[1])
-
-    def _set_credential_based_on_type(self, service_name, key, value):
-        if service_name in key:
-            if self.APIKEY in key:
-                self.set_iam_apikey(value)
-            elif self.URL in key:
-                self.set_url(value)
-            elif self.USERNAME in key:
-                self.username = value
-            elif self.PASSWORD in key:
-                self.password = value
-            elif self.IAM_APIKEY in key:
-                self.set_iam_apikey(value)
-            elif self.IAM_URL in key:
-                self.set_iam_url(value)
-
-    def _load_from_vcap_services(self, service_name):
-        vcap_services = os.getenv('VCAP_SERVICES')
-        if vcap_services is not None:
-            services = json_import.loads(vcap_services)
-            if service_name in services:
-                return services[service_name][0]['credentials']
-        else:
-            return None
+        if self.authenticator:
+            if not isinstance(self.authenticator, Authenticator):
+                raise ValueError(
+                    'authenticator should be of type Authenticator')
```

Comment:
I think you also want to check for if `self.authenticator` is null and throw an error if that's the case

Reply:
The `self.authenticator` can be null initially when initializing the base class. In the generated class we would be doing the check to see if it is `None`